scrapy/utils/url.py

```diff
@@ -103,3 +103,16 @@ def guess_scheme(url):
         return any_to_uri(url)
     else:
         return add_http_if_no_scheme(url)
+
+
+def trim_href_attribute(href):
```

Comment:
The same applies for form action, iframe src, base href, etc.; this seems to be consistent.